spark/src/main/scala/io/delta/tables/execution/DeltaTableOperations.scala

```diff
@@ -112,6 +114,51 @@ trait DeltaTableOperations extends AnalysisHelper { self: DeltaTable =>
     toDataset(sparkSession, restore)
   }
 
+  protected def executeClone(
+    table: DeltaTableV2,
+    target: String,
+    replace: Boolean,
+    tableProperties: Map[String, String],
+    versionAsOf: Option[Long] = None,
+    timestampAsOf: Option[String] = None
+  ): DeltaTable = withActiveSession(sparkSession) {
+    val (targetRelation, targetPath) = if (new Path(target).isAbsolute()) {
+      (UnresolvedRelation(TableIdentifier(target, Some("delta"))), Some(target))
+    } else {
+      val tableIdentifier = sparkSession
+        .sessionState
+        .sqlParser
+        .parseTableIdentifier(target)
+      (UnresolvedRelation(tableIdentifier), None)
+    }
+    val source = DataSourceV2Relation.create(table, None, table.getTableIdentifierIfExists.map(
+      id => Identifier.of(id.database.toArray, id.table)))
+
+    val maybeTimeTravel = if (timestampAsOf.isDefined || versionAsOf.isDefined) {
+      TimeTravel(
+        source,
+        timestampAsOf.map(Literal(_)),
+        versionAsOf,
+        None
+      )
+    } else {
+      source
+    }
+
+    val qe = sparkSession.sessionState.executePlan(CloneTableStatement(maybeTimeTravel,
```

Comment:
We can do 

`val clone = CloneTableStatement(
      timeTraveledSourceRelation,
      targetRelation,
      ifNotExists = false,
      isReplaceCommand = replace,
      isCreateCommand = true,
      tablePropertyOverrides = properties.toMap,
      targetLocation = None)
  toDataset(sparkSession, clone)`
  
  Much cleaner and you don't have to call the SQLExecution and RDD directly.

Reply:
I just copied what table creation was doing since that's what this is now trying to mimic https://github.com/delta-io/delta/blob/master/spark/src/main/scala/io/delta/tables/DeltaTableBuilder.scala#L390

Reply:
@Kimahriman I see, but I think the `toDataset` is pretty neat right? 

This is what we are doing the other `executeF` as well - https://github.com/delta-io/delta/blob/597950c5ff5cf0aa2b76a3d5fba9da58fdba332e/spark/src/main/scala/io/delta/tables/execution/DeltaTableOperations.scala#L47

Btw thank you so much for contributing the Delta Lake btw, really happy to see someone care this much about Delta.

Reply:
@Kimahriman "I just copied what table creation was doing since that's what this is now trying to mimic https://github.com/delta-io/delta/blob/master/spark/src/main/scala/io/delta/tables/DeltaTableBuilder.scala#L390"

Do you have a strong reason why we should reference low-level `rdd`? I think we should try to use only Delta abstractions since it's cleaner.

Reply:
The strong reason is that's what was used in another place in the codebase I was trying to mimic. The other reason is to give the execution a name so you can see it in the UI

Reply:
@Kimahriman If that's the reason then I believe `toDataset(sparkSession, clone)` also does the trick, if we look deeper into how `toDataset` is made.

Reply:
@Kimahriman Also, I just remembered I reviewed a similar issue where the Merge was not visible in the QueryExecutionListener and we fixed by it using `toDataset` 

https://github.com/delta-io/delta/pull/3456
https://github.com/delta-io/delta/issues/1521#issuecomment-1353603522

Reply:
Changed

Reply:
@Kimahriman So you were concerned about whether or not Clone will appear in the Spark's SQL/Dataframe UI right? Just so that I understood you correctly.